_posts/2018-05-15-Rust-turns-three.md

```diff
@@ -0,0 +1,109 @@
+which we’re doing both through conference launches in multiple new continents,
+as well as work toward internationalization throughout the project.
+
+# The Product
+
+If you spend much time reading this blog, you’ll know that the major theme of
+our work over the past year has been **productivity**. As we said
+in [last year’s roadmap](https://blog.rust-lang.org/2017/02/06/roadmap.html):
+
+> From tooling to libraries to documentation to the core language, we want to
+> make it easier to get things done with Rust.
+
+This work will culminate in a major release later this year: **Rust 2018
+Edition**. The release will bring together improvements in every area of the
+project, polished into a new “edition” that bundles the changes together with
+updated documentation and onboarding.
+The [roadmap](https://blog.rust-lang.org/2018/03/12/roadmap.html) has some details about
+what to expect.
+
+The components that make up Rust 2018 will be shipped as they become ready on
+the stable compiler. Recent releases include:
+
+- [On-by-default incremental compilation](https://blog.rust-lang.org/2018/02/15/Rust-1.24.html)
+- [A rewritten official book](https://blog.rust-lang.org/2018/05/10/Rust-1.26.html)
+- [Several of the planned language improvements, including `impl Trait` and `match` improvements](https://blog.rust-lang.org/2018/05/10/Rust-1.26.html)
+
+The next couple of releases will
+include [stable SIMD](https://github.com/rust-lang/rust/issues/48556) support,
+procedural macros, custom allocators, and more. The final big features
+— [lifetime system improvements](https://github.com/rust-lang/rfcs/pull/2094)
+and [async/await](https://github.com/rust-lang/rfcs/pull/2394) — should both
+reach feature complete status on nightly within weeks. Vital tools like
+`rustfmt` are also being polished for the new edition with RFCs for finalizing
+the [style](https://github.com/rust-lang/rfcs/pull/2436)
+and [stability](https://github.com/rust-lang/rfcs/pull/2437) stories. And of
+course the compiler is getting faster, with key benchmarks like Servo’s
+massive [script crate](https://github.com/servo/servo/tree/master/components/script/)
+now well under 30s for `cargo check` after a change.
+
+To help tie all this work to real-world use-cases, we’ve also targeted four
+domains for which Rust provides a compelling end-to-end story that we want to
+show the world as part of Rust 2018. Each domain has a dedicated working group
+and is very much open for new contributors:
+
+- [Embedded devices](https://internals.rust-lang.org/t/announcing-the-embedded-devices-working-group/6839)
+- [Command-line apps](https://internals.rust-lang.org/t/announcing-the-cli-working-group/6872)
+- [The browser, node.js, and portable embedding via WebAssembly](https://internals.rust-lang.org/t/come-join-the-rust-and-webassembly-working-group/6845/2)
+- [Networking services](https://internals.rust-lang.org/t/announcing-the-network-services-working-group-wg-net/7354)
+
```

Comment:
Feels like we may want to give a bit of a mid-year update on those four areas.  Not sure if here or in another post, but I’m sure people will want to know that embedded devices just jumped ahead in capability, as did wasm.